Crashlytics/Crashlytics/Controllers/FIRCLSReportManager.m

```diff
@@ -410,10 +410,18 @@ - (FBLPromise *)deleteUnsentReports {
   NSOperationQueue *__weak queue = _operationQueue;
   FBLPromise *__weak unsentReportsHandled = _unsentReportsHandled;
   promise = [promise then:^id _Nullable(NSNumber *_Nullable value) {
```

Comment:
optional: I wonder if a version below would be a bit more readable?

```
promise = [promise onQueue:dispatch_get_global_queue(QOS_CLASS_UTILITY, 0) then:^id _Nullable(NSNumber *_Nullable value) {
  [queue waitUntilAllOperationsAreFinished];
  return value;
}
.then(^id _Nullable(NSNumber *_Nullable value) {
  [unsentReportsHandled fulfill:nil];
  return value;
})
```

Reply:
I'm happy to make changes that you would like, but I think the GCD folks at Apple would discourage this. Dispatching work to the global concurrent queues that blocks causes dispatch worker threads to be created (for the next block coming through), which is then torn down when it's unblocked. An additional operation on an existing (serial) queue doesn't expand the total concurrency.

Reply:
I do agree with you it's more readable, and it's nice that it doesn't rely on the automatic flatMap behavior of FBLPromise.

Reply:
Good point on blocking a global queue - agreed it's not the best idea. Current version looks good to me from logic perspective. I'll think how we can improve readability and comment if I have any ideas.